qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,532 @@
+
+    Raises:
+      ValueError: if the numbers of operators don't describe a one- or two-body term from
+      an electronic Hamiltonian.
+    """
+    if n_raise == 0 and n_lower == 0:
+        if n_number == 1:
+            return TermType.NUMBER
+        elif n_number == 2:
+            return TermType.COULOMB_EXCHANGE
+        else:
+            raise ValueError("unexpected number of number operators: ", n_number)
+    elif n_raise == 1 and n_lower == 1:
+        if n_number == 1:
+            return TermType.NUMBER_EXCITATION
+        elif n_number == 0:
+            return TermType.EXCITATION
+        else:
+            raise ValueError("unexpected number of number operators: ", n_number)
+    elif n_raise == 2 and n_lower == 2:
+        return TermType.DOUBLE_EXCITATION
+    else:
+        raise ValueError("unexpected number of operators: ", n_number)
+
+
+def _get_adjacency_matrix(fer_op: FermionicOp) -> np.ndarray:
+    """
+    Return an adjacency matrix specifying the edges in the BKSF graph for the
+    operator `fer_op`.
+
+    The graph is undirected, so we choose to return the edges in the upper triangle.
+    (There are no self edges.). The lower triangle are all `False`.
+
+    Returns:
+          numpy.ndarray(dtype=bool): edge_matrix the adjacency matrix.
+    """
+    n_modes = fer_op.register_length
+    edge_matrix = np.zeros((n_modes, n_modes), dtype=bool)
+    for term in fer_op.to_list():
+        _add_edges_for_term(edge_matrix, _operator_string(term))
+    return edge_matrix
+
+
+def _add_one_edge(edge_matrix: np.ndarray, i: int, j: int) -> None:
+    """
+    Add an edge from lesser index to greater. This maintains the upper triangular structure.
+    """
+    if i == j:
+        raise ValueError("expecting i != j")
+    edge_matrix[min(i, j), max(i, j)] = True
+
+
+def _add_edges_for_term(edge_matrix: np.ndarray, term_str: str) -> None:
+    """
+    Add one, two, or no edges to `edge_matrix` as dictated by the operator `term_str`.
+    """
+    (n_number, n_raise, n_lower), facs = _unpack_term(term_str)
+    _type = _interaction_type(n_number, n_raise, n_lower)
+    # For EXCITATION and NUMBER_EXCITATION, create and edge between the `+` and `-`.
+    if _type in (TermType.EXCITATION, TermType.NUMBER_EXCITATION):
+        inds = [i for (i, c) in facs if c in "+-"]
+        if len(inds) != 2:
+            raise ValueError("wrong number or raising and lowering: ", len(inds))
+        _add_one_edge(edge_matrix, *inds)
+    # For `double_excitation` create an edge between the two `+`s and edge between the two `-`s.
+    elif _type == TermType.DOUBLE_EXCITATION:
+        raise_inds = [i for (i, c) in facs if c == "+"]
+        lower_inds = [i for (i, c) in facs if c == "-"]
+        _add_one_edge(edge_matrix, *raise_inds)
+        _add_one_edge(edge_matrix, *lower_inds)
+
+
+def bksf_edge_list_fermionic_op(ferm_op: FermionicOp) -> np.ndarray:
+    """
+    Construct edge list required for the bksf algorithm.
```

Comment:
I would stick to `BKSF` in all-caps everywhere.

Reply:
Done